### PR TITLE
Test unlockwalletguard

### PR DESCRIPTION
**Pull request by johnedma**
Soundtrack of this PR: [The Notorious B.I.G. ft. Frank Sinatra - Blue Eyes Meets Bed-Stuy | FULL ALBUM](https://www.youtube.com/watch?v=LSPUO4Xc4N8)

### Motivation

Continued test coverage expansion for Desktop Wallet components, views, utils and hooks.

### In this PR

- Tests: 
  - unauthenticated with no entropy string 
  - authenticated with no entropy string
  - authenticated with entropy string 
  - unauthenticated with entropy string 
- Discovered auth edge case that required updating AuthFlowGuard to prevent future bugs

### Testing

 PASS  test/components/UnlockWalletGuard/UnlockWalletGuard.test.tsx (13.384 s)
File                  | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
----------------------|---------|----------|---------|---------|-------------------
UnlockWalletGuard.tsx                        |     100 |      100 |     100 |     100 |          

Test Suites: 1 passed, 1 total
Tests:       4 passed, 4 total
Snapshots:   0 total
Time:        21.486 s

### CODE REVIEW

File: test/components/UnlockWalletGuard/UnlockWalletGuard.test.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    expect(screen.queryByText('Create a new account for this desktop wallet.')).toBeInTheDocument();
+    expect(children).not.toBeInTheDocument();
+  });
+
+  test('authenticated with no entropy string', () => {
+  // @ts-ignore mock
+    const { children } = setupComponent({ encryptedEntropy: null, isAuthenticated: true });
+
+    expect(screen.queryByTestId('DashboardOverview')).not.toBeInTheDocument();
```
Comment (davidyhall) [inline test/components/UnlockWalletGuard/UnlockWalletGuard.test.tsx:44]: Just fyi, you could also pull the `screen.queryByTestId('DashboardOverview')` into `setupComponent`. But it doesn't really do anything to the code.

**Review by davidyhall — APPROVED** [review]
LGTM
